aten/src/ATen/native/sparse/ValidateCompressedIndicesCommon.h

```diff
@@ -241,52 +272,75 @@ void _validate_compressed_sparse_indices_kernel(
 
     const auto batch_dims = cidx.sizes().slice(0, cidx.dim() - 1);
     const auto batch_count = indexCount(batch_dims);
-    const auto batch_idx = at::arange(batch_count, cidx.options()).view(batch_dims).unsqueeze_(-1);
+    const auto batch_idx =
+        at::arange(batch_count, cidx.options()).view(batch_dims).unsqueeze_(-1);
+
+    const auto idx_ndims = idx.dim();
+    std::array<int64_t, MAX_DIMS> idx_sizes, idx_strides;
+    for (int i = 0; i < idx_ndims; i++) {
```

Comment:
We should probably error out if `idx_ndims > MAX_DIMS`, otherwise it is a UB.

Reply:
Fixed: sizes and strides are now passed to the kernel via `Tensor` instance and we don't depend on `MAX_DIMS` anymore.